zerver/lib/actions.py

```diff
@@ -6398,6 +6400,11 @@ def do_update_message_flags(
         "messages": messages,
         "all": False,
     }
+
+    if flag == "read" and operation == "remove":
+        raw_unread_data = get_raw_unread_data(user_profile, messages)
+        event["message_info_dict"] = send_unread_msgs(raw_unread_data)
```

Comment:
We should rename `send_unread_msgs` to something like `format_unread_message_update`

Reply:
We might also want `s/message_info_dict/message_details/`.

Reply:
Went with `message_details` and `format_unread_message_details`.